src/typesystem.hpp

```diff
@@ -36,7 +36,7 @@ using TypeFunctionStructure = std::unordered_map<std::string, MonoId>;
 struct TypeFunction {
 	int argument_count; // -1 means variadic
 	TypeFunctionStructure structure; // can be nullptr
-	bool is_dummy;
+	bool is_dummy = false;
```

Comment:
We should really use brace notation for default values. It's just more idiomatic

 ```c++
bool is_dummy { false };
```